src/app.js

```diff
@@ -1,7 +1,20 @@
 /* @flow */
 'use strict'
 
+// SIGINT and uncaughtException do not run cleanup
+// so we have an exitHandler to clean up the database connection
+function exitHandler (error) {
+  if (error) {
+    console.error('Exiting: ', error)
+    process.exit(1)
+  }
+  process.exit(0)
+}
+
 module.exports = require('./lib/app')
 if (!module.parent) {
+  process.on('SIGINT', exitHandler)
+  process.on('uncaughtException', exitHandler)
```

Comment:
> Oh, so just call knex.destroy().

I've tried this and it didn't work locally. I'll test again. 

> If you don't call process.exit(), then you're not preventing the default handler you're just adding something to it.

I don't think I understand what you mean by "default handler". SIGINT is sent to a process when a user presses ctrl+c. A process catches SIGINT to be able to gracefully shutdown. I don't think theres anything to do other than exiting after cleaning up in the event handler. There are no default handlers.